### ISSUE TITLE
Add support for dtype=object for backend tests

### ISSUE THREAD

**Issue by hamptonm1** [onnx/onnx-mlir#2416]
For the Equal Operation, the onnx string type backend tests are expecting `np.object` for the input. We currently do not have support for this data type and therefore the two backend tests for the Equal Operation have been commented out.

"test_equal_string_broadcast_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},
"test_equal_string_cpu": {STATIC_SHAPE:{}, DYNAMIC_SHAPE:{-1:{-1}}, CONSTANT_INPUT:{-1}},

For more information please see the following PR: https://github.com/onnx/onnx-mlir/pull/2401

**Comment by hamptonm1** [comment]
I tried updating `onnx-mlir/src/Runtime/PyExecutionSessionBase.cpp` with the following:

```
 // string type missing
    else if (py::isinstance<py::array_t<const char*>>(inputPyArray))
      dtype = ONNX_TYPE_STRING;
```

And then I get this error:

`RuntimeError: NumPy type info missing for PKc`
PKc maps to {"PKc", ONNX_TYPE_STRING},    // const char * -> STRING. So although we have the support for string, seems like numpy keeps complaining that it was not correctly implemented.

**Comment by chentong319** [comment]
I think this issue is the same as  #2164, though they emphasizes different parts of the problem.

**Comment by hamptonm1** [comment]
> I think this issue is the same as #2164, though they emphasizes different parts of the problem.

Agreed....similar but I just opened it anyways to keep track so that the backend tests are eventually enabled.

**Comment by chentong319** [comment]
> I tried updating `onnx-mlir/src/Runtime/PyExecutionSessionBase.cpp` with the following:
> 
> ```
>  // string type missing
>     else if (py::isinstance<py::array_t<const char*>>(inputPyArray))
>       dtype = ONNX_TYPE_STRING;
> ```
> 

The error came from `py::isinstance<py::array_t<const char*>>(inputPyArray)`

**Comment by hamptonm1** [comment]
> > I tried updating `onnx-mlir/src/Runtime/PyExecutionSessionBase.cpp` with the following:
> > ```
> >  // string type missing
> >     else if (py::isinstance<py::array_t<const char*>>(inputPyArray))
> >       dtype = ONNX_TYPE_STRING;
> > ```
> 
> The error came from `py::isinstance<py::array_t<const char*>>(inputPyArray)`

Yup I already tried that in the PR I originally linked. I specified that I tried this and also I added the error message I received. Which points to pKc or ONNX_TYPE_STRING
